client/packages/system/src/Patient/Encounter/CreateEncounterModal.tsx

```diff
@@ -33,6 +34,47 @@ import {
   ClinicianSearchInput,
 } from '../../Clinician';
 import { PatientTabValue } from '../PatientView/PatientView';
+import { PickersDay, PickersDayProps } from '@mui/x-date-pickers';
+import Badge from '@mui/material/Badge';
+
+type HighlightedDay = { datetime: Date; label?: string | null };
+type BadgePickersDayProps = {
+  highlightedDays: HighlightedDay[];
+};
+const BadgePickersDay = (
+  props: PickersDayProps<Date> & BadgePickersDayProps
+) => {
+  const { highlightedDays = [], day, outsideCurrentMonth, ...other } = props;
+
+  const matchingDay = highlightedDays.find(it => {
+    const date = it.datetime;
+    return (
+      day.getFullYear() === date.getFullYear() &&
+      day.getMonth() === date.getMonth() &&
+      day.getDate() === date.getDate()
+    );
+  });
```

Comment:
adding in the date-fns `isSameDay` function to DateUtils will allow the simpler:
```suggestion
  const matchingDay = highlightedDays.find(it =>
    DateUtils.isSameDay(it.datetime, day)
  );
```